src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/POPUserMetricsService.kt

```diff
@@ -1,30 +1,98 @@
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.config.ResourceNotFoundException
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.LicenceTag
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.PopUserCountMetric
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.PopUserCountMetrics
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.service.external.PrisonRegisterApiService
 
 @Service
 class POPUserMetricsService(
   private val registry: MeterRegistry,
-  private val prisonOTPService: PoPUserOTPService,
+  private val popUserOTPService: PoPUserOTPService,
+  private val prisonRegisterApiService: PrisonRegisterApiService,
+  private val licenceConditionService: LicenceConditionService,
 ) {
+  private val popUserCountMetrics = PopUserCountMetrics()
 
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
   fun recordCustomMetrics() {
-    recordProbationUsersMetrics()
+    recordProbationUsersLicenceConditionMetrics()
   }
 
-  fun recordProbationUsersMetrics(): Int {
-    val otpList = prisonOTPService.getAllOTPs()
-    val cnt = otpList?.size ?: 0
-    registry.gauge(
-      "total_pack_print_count",
-      cnt,
-    )
-    return cnt
+  fun recordProbationUsersLicenceConditionMetrics() {
+    log.info("Started running scheduled POP User metrics job - LicenceCondition")
+    val popUserList = popUserOTPService.getAllOTPs()
+
+    var totalPopUser = 0
+    if (popUserList != null) {
+      totalPopUser = popUserList.size
+    }
+    val prisonList = prisonRegisterApiService.getActivePrisonsList()
+    if (popUserList != null) {
+      var percentStdLicenceCondition: Int
+      var percentOtherLicenceCondition: Int
+      for (prison in prisonList) {
+        var stdLicenceConditionCount = 0
+        var otherLicenceConditionCount = 0
+
+        try {
+          for (popUser in popUserList) {
+            if (popUser.prisoner.prisonId.equals(prison.id)) {
+              val licencesConditions = licenceConditionService.getLicenceConditionsByNomsId(popUser.prisoner.nomsId)
+              val stdLicenceConditionList = licencesConditions?.standardLicenceConditions
+              val otherLicenceConditionList = licencesConditions?.otherLicenseConditions
+              if (!stdLicenceConditionList.isNullOrEmpty()) {
+                stdLicenceConditionCount += 1
+              }
+              if (!otherLicenceConditionList.isNullOrEmpty()) {
+                otherLicenceConditionCount += 1
+              }
+            }
+          }
+          percentStdLicenceCondition = if (stdLicenceConditionCount > 0) {
+            100 - ((stdLicenceConditionCount / totalPopUser) * 100)
+          } else {
+            100
+          }
+
+          percentOtherLicenceCondition = if (otherLicenceConditionCount > 0) {
```

Comment:
this could be easier to refactor as 
`fun calculatePercentage(count: Int, total: Int): Double {
    return if (count > 0) {
        100 - ((count.toDouble() / total) * 100)
    } else {
        100
    }`

Reply:
done